myoFlex/myoFlex.ino

```diff
@@ -1,69 +1,151 @@
+  pointer.attach(pointerPin, min, max);
+  middle.attach(middlePin, min, max);
+  ring.attach(ringPin, min, max);
+  pinky.attach(pinkyPin, min, max);
+  Serial.println("Starting");       //makes the hand relax all of the way
+  handPosition(max,max,max,max,max);
+  trigger = setTrigger();
+}
+
+//setup starts by linking the Servo objects to each pin and detailing the limits per each argument. Then it opens the hand for a default position and will call the setTrigger function. Once it's collected, the setup is complete
+
+void loop() {
+  Serial.println(analogRead(myoIn));
+  if(check(currentVoltage = analogRead(myoIn))){//collects voltage, then assigns to currentVoltage, then passes currentVoltage to check function, then checks the check return value to see if it's higher than trigger value
+    state++; //increases our state
+    state = state%4; //ensoures our only values are between 0-3 and loops back if otherwise
+    Serial.print("state is ");
+    Serial.println(state);
+    //Serial.println(currentVoltage);        //prints in the serial moniter
+    if(state == 0){
+      Serial.println("Open");
+      handPosition(max,max,max,max,max);
+    }
+    else if(state == 1){
+      Serial.println("Point");
+      handPosition(max,min,max,max,max);
+    }
+    else if(state == 2){
+      Serial.println("Pinch");
+      handPosition(min,min,min,max,max);
+    }
+    else if(state == 3){
+      Serial.println("Fist");
+      handPosition(min,min,min,min,min);
+    }
+    else {
+      state = 1; //edgecase if logic fail
+      handPosition(max,max,max,max,max);
+    }
+  }
+}
+
+
+//Finds the max flex for a total of 3 trials
+int setTrigger(){
+
+  float average;
+  do{
+    float sumMaxFlexOfTrials = 0; //the sum of the max reading found for each of the three trials
+    
+    for(int j = 1; j <= 3; j++){  //loops to collect the max of the three trials
+      Serial.print("Reading max of Flex ");
+      Serial.print(j);
+      Serial.println(" reading during the next 3 seconds");
+
+      int intervalOfEachTrial = 3000; //interval time in milliseconds that each trial happens
+      int intervalOfReadings = 10; //interval to compare readings in milliseconds
+      float maxRecordedFlex = 0;
+      
+      //collects the first two readings of the trial
+      float reading1 = analogRead(myoIn);
+      delay(intervalOfReadings);
+      float reading2 = analogRead(myoIn);
+      delay(intervalOfReadings);
+
+      int remainderOfTrial = intervalOfEachTrial - intervalOfReadings;
+      for(int i = remainderOfTrial; i > 0; i -= intervalOfReadings){  //for the remainder of the trial collects the next data point and checks if the previous data point was the absolute max
+        float reading3 = analogRead(myoIn); 
+        float currentReading = reading2;
+        
+        if(currentReading > reading1 && currentReading < reading3 && currentReading > maxRecordedFlex) //checks if currentReading is a relative max, then checks if it is the absolute max
```

Comment:
I think instead of maxRecordedFlex we should use a simple threshold, say 30% of the maximum value, just so we can make sure it is a true spike? if we use maxRecordedFlex, we might have trouble hitting a really high, but still legitmate flex. Thoughts